packages/astro/src/runtime/server/render/util.ts

```diff
@@ -104,6 +104,11 @@ Make sure to use the static attribute syntax (\`${key}={value}\`) instead of the
 		return markHTMLString(` class="${toAttributeString(value, shouldEscape)}"`);
 	}
 
+	// Prevents URLs in attributes from being escaped in static builds
+	if (typeof value === 'string' && value.includes('&') && URL.canParse(value)) {
```

Comment:
I noticed that [`URL.canParse`](https://nodejs.org/api/url.html#urlcanparseinput-base) was added in Node v18.17.0, but our minimum required version is still v18.14.1.

We should either implement a fallback for `URL.canParse` (`try/catch`) or wait on merging this until we bump our engines to v.18.17.0 (possibly in the next minor cc @Princesseuh)

Reply:
[`URL.canParse`](https://url.spec.whatwg.org/#dom-url-canparse) uses the same internal method as the [`URL constructor`](https://url.spec.whatwg.org/#constructors), so a try-catch should be sufficient to unblock.